core/src/test/scala/unit/kafka/server/KafkaApisTest.scala

```diff
@@ -5289,6 +5304,156 @@ class KafkaApisTest extends Logging {
+      any()
+    )).thenReturn(Errors.NONE)
+    when(sharePartitionManager.newContext(any(), any(), any(), any(), any())).thenReturn(
+      new FinalContext(new mutable.HashMap[TopicIdPartition, ShareFetchRequest.SharePartitionData]().asJava)
+    )
+
+    when(clientQuotaManager.maybeRecordAndGetThrottleTimeMs(
+      any[RequestChannel.Request](), anyDouble, anyLong)).thenReturn(0)
+
+    when(sharePartitionManager.acknowledge(
+      anyString(),
+      anyString(),
+      any()
+    )).thenReturn(CompletableFuture.completedFuture(Map[TopicIdPartition, ShareAcknowledgeResponseData.PartitionData](
+      new TopicIdPartition(topicId, new TopicPartition(topicName, 0)) ->
+        new ShareAcknowledgeResponseData.PartitionData()
+          .setPartitionIndex(0)
+          .setErrorCode(Errors.NONE.code())
+    ).asJava))
+
+    when(sharePartitionManager.releaseAcquiredRecords(
+      anyString(),
+      anyString(),
+      any()
+    )).thenReturn(CompletableFuture.completedFuture(Map[TopicIdPartition, ShareAcknowledgeResponseData.PartitionData](
+      new TopicIdPartition(topicId, new TopicPartition(topicName, 0)) ->
+        new ShareAcknowledgeResponseData.PartitionData()
+          .setPartitionIndex(0)
+          .setErrorCode(Errors.NONE.code()),
+      new TopicIdPartition(Uuid.randomUuid(), new TopicPartition("bar", 0)) ->
+        new ShareAcknowledgeResponseData.PartitionData()
+          .setPartitionIndex(0)
+          .setErrorCode(Errors.INVALID_RECORD_STATE.code()),
+      new TopicIdPartition(topicId, new TopicPartition(topicName, 0)) ->
+        new ShareAcknowledgeResponseData.PartitionData()
+          .setPartitionIndex(0)
+          .setErrorCode(Errors.NONE.code())
+    ).asJava))
+
+    val shareFetchRequestData = new ShareFetchRequestData().
+      setGroupId("group").
+      setMemberId(memberId.toString).
+      setShareSessionEpoch(ShareFetchMetadata.FINAL_EPOCH).
+      setTopics(List(new ShareFetchRequestData.FetchTopic().
+        setTopicId(topicId).
+        setPartitions(List(
+          new ShareFetchRequestData.FetchPartition()
+            .setPartitionIndex(0)
+            .setPartitionMaxBytes(0)
+            .setAcknowledgementBatches(List(
+              new ShareFetchRequestData.AcknowledgementBatch()
+                .setBaseOffset(11)
+                .setLastOffset(20)
+            ).asJava)
+        ).asJava)
+      ).asJava)
+
+    val shareFetchRequest = new ShareFetchRequest.Builder(shareFetchRequestData).build(ApiKeys.SHARE_FETCH.latestVersion)
+    val request = buildRequest(shareFetchRequest)
+    kafkaApis = createKafkaApis(overrideProperties = Map(KafkaConfig.ShareGroupEnableProp -> "true"))
+    kafkaApis.handleShareFetchRequest(request)
+    val response = verifyNoThrottling[ShareFetchResponse](request)
+    val responseData = response.data()
+    val topicResponses = response.data().responses()
+
+    assertEquals(Errors.INVALID_RECORD_STATE.code(), responseData.errorCode())
```

Comment:
Wouldn't this actually be `INVALID_RECORD_STATE` in the acknowledge error code, and then `NONE` for the error code? This is a piggyback acknowledge which is in error.

Reply:
In this case, the `acknowledge()` function is mocked to return a successful acknowledgement [here](https://github.com/confluentinc/kafka/pull/1149/files#diff-d418e862d65bd68453ec359a3b4f08e38be8948b4d772909ece6ff2a07e5f714R5327) and  a faulty response is mocked on calling `releaseAcquiredRecords()` [here](https://github.com/confluentinc/kafka/pull/1149/files#diff-d418e862d65bd68453ec359a3b4f08e38be8948b4d772909ece6ff2a07e5f714R5338). Hence, the acknowledgements will be processed successfully and there will be top level error code when releasing acquired records [here](https://github.com/confluentinc/kafka/pull/1149/files#diff-cc056b4960ededba37a438b1454f0f3c5ff5e8ad5e6d2ec9a08e813ca056ffebR1688)

Reply:
I think you should swallow the error in this case. If the session is valid, but actually when we try to clean up the acquired records, we hit an error, the client can't do anything about it.

Reply:
I wonder what @apoorvmittal10 thinks.

Reply:
yeah, that does make sense. i can remove the top level error code in the case where we end up having some error while releasing the acquired records.

Reply:
I have modified the code to not return any error to the client while releasing acquired records on session close (given the session is valid)